# --- T2-COPYRIGHT-NOTE-BEGIN ---
# T2 SDE: package/*/xfsprogs/hotfix-noattr.patch
# Copyright (C) 2024 The T2 SDE Project
# 
# This Copyright note is generated by scripts/Create-CopyPatch,
# more information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License version 2 as used by the T2 SDE.
# --- T2-COPYRIGHT-NOTE-END ---

--- xfsprogs-6.10.1/io/Makefile.vanilla	2024-09-30 21:27:37.706000000 +0100
+++ xfsprogs-6.10.1/io/Makefile	2024-09-30 21:28:33.695000000 +0100
@@ -20,7 +20,6 @@
 	fiemap.c \
 	file.c \
 	freeze.c \
-	fsproperties.c \
 	fsuuid.c \
 	fsync.c \
 	getrusage.c \
@@ -51,6 +50,11 @@
 	truncate.c \
 	utimes.c
 
+ifeq ($(HAVE_LIBATTR),yes)
+	CFILES+=fsproperties.c
+	CFLAGS += -DHAVE_LIBATTR
+endif
+
 LLDLIBS = $(LIBXCMD) $(LIBHANDLE) $(LIBFROG) $(LIBPTHREAD) $(LIBUUID)
 LTDEPENDENCIES = $(LIBXCMD) $(LIBHANDLE) $(LIBFROG)
 LLDFLAGS = -static-libtool-libs
--- xfsprogs-6.10.1/io/init.c.vanilla	2024-09-30 21:28:55.115000000 +0100
+++ xfsprogs-6.10.1/io/init.c	2024-09-30 21:29:47.044000000 +0100
@@ -89,7 +89,9 @@
 	utimes_init();
 	crc32cselftest_init();
 	exchangerange_init();
+#ifdef HAVE_LIBATTR
 	fsprops_init();
+#endif
 }
 
 /*
--- xfsprogs-6.10.1/scrub/phase1.c.vanilla	2024-09-30 21:30:06.969000000 +0100
+++ xfsprogs-6.10.1/scrub/phase1.c	2024-09-30 21:36:07.175000000 +0100
@@ -141,6 +141,7 @@
 mode_from_autofsck(
 	struct scrub_ctx	*ctx)
 {
+#ifdef HAVE_LIBATTR
 	struct fsprops_handle	fph = { };
 	char			valuebuf[FSPROP_MAX_VALUELEN + 1] = { 0 };
 	size_t			valuelen = FSPROP_MAX_VALUELEN;
@@ -177,6 +178,9 @@
 	}
 
 	fsprops_free_handle(&fph);
+#else
+	goto no_property;
+#endif
 
 summarize:
 	switch (ctx->mode) {
